## Домашнє завдання 6: Статистичні тести

0. Зчитайте дані з `data.csv` та запишіть значення колонки data у форматі np.ndarray в змінну data. Ми будемо далі працювати з цією змінною.

In [1]:
from google.colab import drive
drive.mount('/drive')

Mounted at /drive


In [4]:
import pandas as pd
import numpy as np

# Step 1. Read CSV file
df = pd.read_csv('/drive/MyDrive/Colab_Notebooks/Data_Analyst_Course_HP/Module_6/Data_M6/data.csv')

# Step 2. Save column 'data' as numpy array
data = df['data'].to_numpy()


Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного вектора, тобто мати розмір (100,).

In [5]:
data.shape

(100,)

1. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2022 року.

До цього місяця, ми мали продажі в середньому на чек $20. Чи вийшло у нас статистично значущо **підвищити** середній чек за останній місяць?

Оберіть статистичний тест (z-test чи t-test). Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чеку.

**Підказка**: тут тест буде одновибірковий.

In [7]:
from scipy import stats

# Step 1. The known value of the average check up to this month
mu0 = 20

# Step 2. Conduct a one-sample t-test
t_stat, p_value = stats.ttest_1samp(data, mu0)

print(f'p-value: {p_value}')
print(f'Average value of the sample: {np.mean(data):.2f}')


p-value: 0.3219937839011988
Average value of the sample: 19.38


Conclusion:

1. Since p-value (0.322) is greater than the significance level of 0.05, we cannot reject the null hypothesis.
2. Although we see a slight decrease in the average check from 20 USD to 19.38 USD, this difference is not statistically significant.
3. This means that any observed differences are likely due to random variation rather than a real change in customer behavior.

4. We did not achieve a statistically significant change in the average check value this month.

2. Пройшов час, і ми зібрали дані про продажі за серпень 2022 року, які зберігли у файл `data_2.csv`. Продажів було дещо більше, але чи змінився стат. значущо середній чек порівняно з липнем 2022? Перевірте це використовуючи ztest.

  Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями.

  **Підказка**: треба провести двовибірковий ztest з Н0 - що середні значення в липні та серпні 2022 року - однакові. Для кращого розуміння ситуації, можливо вам буде цікаво переглянути, які середні показники чеку були в кожен з місяців. Порівняйте свій висновок з простого спостереження середніх значень і висновок статистичного тесту. Чи вони збігаються?

In [8]:
import pandas as pd
import numpy as np
from statsmodels.stats.weightstats import ztest

# Step 1. Read CSV files
df1 = pd.read_csv('/drive/MyDrive/Colab_Notebooks/Data_Analyst_Course_HP/Module_6/Data_M6/data.csv')
df2 = pd.read_csv('/drive/MyDrive/Colab_Notebooks/Data_Analyst_Course_HP/Module_6/Data_M6/data_2.csv')

# Step 2. Convert to numpy arrays
july_data = df1['data'].to_numpy()
august_data = df2['data'].to_numpy()

# Step 3. Calculate means for both months
print(f'July mean: {np.mean(july_data):.2f}')
print(f'August mean: {np.mean(august_data):.2f}')

# Step 4. Perform two-sample z-test
z_stat, p_value = ztest(july_data, august_data)
print(f'p-value: {p_value}')



July mean: 19.38
August mean: 19.37
p-value: 0.9958137794593356


**Conclusion:**

Statistical test results:

p-value (0.9859) is much larger than significance level (0.05)
This strongly suggests we cannot reject the null hypothesis
There is no statistically significant difference between the means.